### PR TITLE
Feature/edit delete project

### PR DESCRIPTION
**Pull request by Jaranag**
· Added edit and delete project functions to font awesome icons in the project component
· Deleting project basing on id
· To edit project we now change isEditing to true and show an input intead of the project component
· This input's placeholder is the project name
· Using filter function to return new arrays but without the project with a selected id
· Same for edit but returning with project updated

### CODE REVIEW

**Review by JiajunQiuWu — APPROVED** [review]
eres una maquina

**Review by rekloyd — APPROVED** [review]

**Review by JoseAlfredoMiguel — APPROVED** [review]

File: src/components/ProjectComponent.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faPenToSquare } from '@fortawesome/free-solid-svg-icons'
 import { faTrash } from '@fortawesome/free-solid-svg-icons'
-import { ProjectComponentProps } from '../types'
 import React from 'react'
+import { Project } from '../types'
 
 export const ProjectComponent: React.FC<ProjectComponentProps> = ({project,deleteproject,editproject}) => {
```
Comment (SaraVictorFernandez) [inline src/components/ProjectComponent.tsx:7]: Better to say that {project, deleteproject,editproject} implements interface ProjectComponentProps

File: src/components/ProjectForm.tsx [hunk 2]
```diff
@@ -26,6 +26,10 @@ export const ProjectForm: React.FC<ProjectFormProps> = ({ addProject }) => {
   )
 } 
 
+export interface ProjectFormProps {
```
Comment (SaraVictorFernandez) [inline src/components/ProjectForm.tsx:29]: duplicated

File: src/components/Sidebar.tsx [hunk 3]
```diff
@@ -19,22 +20,46 @@ export const Sidebar = () => {
...
+      projects.map((project) =>
+        project.id === id ? { ...project, isEditing: !project.isEditing } : project
+      )
+    );
+  };
+
+  const editProjectName = (projectName: string, id: string) => {
+    setProjects(
+      projects.map((p) =>
+        p.id === id? {...p, projectName, isEditing:!p.isEditing } : p
```
Comment (SaraVictorFernandez) [inline src/components/Sidebar.tsx:40]: assigning `isEditing: !p.isEditing` can create bugs in the future

When you edit a project name, you know for a fact that you are not editing anymore. you can directly set `isEditing: false`

Comment (rekloyd) [reply src/components/Sidebar.tsx]: It is changed

File: src/components/Sidebar.tsx [hunk 4]
```diff
@@ -19,22 +20,46 @@ export const Sidebar = () => {
     setProjects([...projects, newProject]);
   };
 
+  const deleteproject = (id: string) => {
+
+    setProjects(projects.filter(project => project.id !== id))
 
+  }
+
+  const editproject = (id: string) => {
```
Comment (SaraVictorFernandez) [inline src/components/Sidebar.tsx:29]: Maybe we could use a better name like "startEditingProject"

**Review by SaraVictorFernandez — APPROVED** [review]
looks good to me

**Pull request merged**